src/components/DataExplorerFrame.js

```diff
@@ -9,32 +9,32 @@ import * as Nav from 'src/libs/nav'
 const DataExplorerFrame = ({ dataset }) => {
   const elem = useRef()
   const { origin } = _.find({ name: dataset }, datasets)
-
-  useEffect(() => {
-    iframeResizer({
-      onMessage: ({ message: { importDataQueryStr, deQueryStr } }) => {
-        if (importDataQueryStr) {
-          Nav.history.push({
-            pathname: Nav.getPath('import-data'),
-            search: `?${importDataQueryStr}`
-          })
-        } else if (deQueryStr) {
-          // Propagate Data Explorer URL params to app.terra.bio.
-          // Don't call Nav.history.replace(). That will trigger a request and
-          // cause the page to flicker.
-          window.history.replaceState({}, `Data Explorer - ${dataset}`, `#${Nav.history.location.pathname.slice(1)}?${deQueryStr}`)
-        }
-      }
-    }, elem.current)
-
-    return elem.current.iFrameResizer.removeListeners
-  }, [dataset])
-
   return iframe({
     src: `${origin}/?embed&${Nav.history.location.search.slice(1)}`,
     ref: elem,
     style: { border: 'none', width: '100%' },
-    title: `${dataset} - Data Explorer`
+    title: `${dataset} - Data Explorer`,
+    display: 'block',
+    width: '100%',
+    onLoad: () => {
+      iframeResizer({
+        onMessage: ({ message: { importDataQueryStr, deQueryStr } }) => {
+          if (importDataQueryStr) {
+            Nav.history.push({
+              pathname: Nav.getPath('import-data'),
+              search: `?${importDataQueryStr}`
+            })
+          } else if (deQueryStr) {
+            // Propagate Data Explorer URL params to app.terra.bio.
+            // Don't call Nav.history.replace(). That will trigger a request and
+            // cause the page to flicker.
+            window.history.replaceState({}, `Data Explorer - ${dataset}`, `#${Nav.history.location.pathname.slice(1)}?${deQueryStr}`)
+          }
+        }
+      }, elem.current)
+
+      return elem.current.iFrameResizer.removeListeners
```

Comment:
Returning this from `useEffect` causes it to be run on unmount; however, doing so from here doesn't. You'll need to set this up to run on unmount.